src/main/java/hudson/plugins/jira/JiraBuildAction.java

```diff
@@ -22,68 +20,68 @@
 @ExportedBean
 public class JiraBuildAction implements Action, RunAction2, Serializable {
 
-    private transient Run<?, ?> owner;
-
-    private Set<JiraIssue> issues;
-
-    @Override
-    public void onAttached(Run<?, ?> r) {
-        this.owner = r;
-    }
-
-    @Override
-    public void onLoad(Run<?, ?> r) {
-        this.owner = r;
-    }
-
-    public JiraBuildAction(@Nonnull Set<JiraIssue> issues) {
-        this.issues = new HashSet( issues);
-    }
-
-    public String getIconFileName() {
-        return null;
-    }
-
-    public String getDisplayName() {
-        return Messages.JiraBuildAction_DisplayName();
-    }
-
-    public String getUrlName() {
-        return "jira";
-    }
-
-    public Run<?, ?> getOwner() {
-        return owner;
-    }
-
-    @Exported(inline = true)
-    public Set<JiraIssue> getIssues() {
-        return issues;
-    }
-
-    @Exported
-    public String getServerURL() {
-        JiraSite jiraSite = JiraSite.get(owner.getParent());
-        URL url = jiraSite != null ? jiraSite.getUrl() : null;
-        return url != null ? url.toString() : null;
-    }
-
-    /**
-     * Finds {@link JiraIssue} whose ID matches the given one.
-     * @param issueID e.g. JENKINS-1234
-     * @return JIRAIssue representing the issueID
-     */
-    public JiraIssue getIssue(String issueID) {
-        for (JiraIssue issue : issues) {
-            if (issue.getKey().equals(issueID)) {
-                return issue;
-            }
-        }
-        return null;
-    }
-
-    public void addIssues(Set<JiraIssue> issuesToBeSaved) {
-        this.issues.addAll(issuesToBeSaved);
+  private static final long serialVersionUID = 457861802965070978L;
+  private final transient Set<JiraIssue> issues;
```

Comment:
`transient` means it will never be saved! That is not what you want.

Sonar is just being dumb—it does not notice that the implementation class `HashSet` _is_ `Serializable`.
```suggestion
  private final HashSet<JiraIssue> issues;
```

Reply:
Yeah, I was just looking how to get CI happy temporarilly, but this is also coming from SpotBugs. I'll revert to non-implementing `Serializable` and see how it works then.

Reply:
Oh right, there is no need to implement `Serializable` on `JiraBuildAction`; XStream will ignore that. (Same for the `serialVersionUID`.)